beacon_chain/utils/bls/src/signature.rs

```diff
@@ -0,0 +1,81 @@
+use super::ssz::{decode_ssz_list, Decodable, DecodeError, Encodable, SszStream};
+use bls_aggregates::{PublicKey, SecretKey, Signature as RawSignature};
+
+/// A single BLS signature.
+///
+/// This struct is a wrapper upon a base type and provides helper functions (e.g., SSZ
+/// serialization).
+#[derive(Debug, PartialEq, Clone)]
+pub struct Signature(RawSignature);
+
+impl Signature {
+    /// Instantiate a new Signature from a message and a SecretKey.
+    pub fn new(msg: &[u8], sk: &SecretKey) -> Self {
+        Signature(RawSignature::new(msg, sk))
+    }
+
+    /// Instantiate a new Signature from a message and a SecretKey, where the message has already
+    /// been hashed.
+    pub fn new_hashed(msg_hashed: &[u8], sk: &SecretKey) -> Self {
+        Signature(RawSignature::new_hashed(msg_hashed, sk))
+    }
+
+    /// Verify the Signature against a PublicKey.
+    pub fn verify(&self, msg: &[u8], pk: &PublicKey) -> bool {
+        self.0.verify(msg, pk)
+    }
+
+    /// Verify the Signature against a PublicKey, where the message has already been hashed.
+    pub fn verify_hashed(&self, msg_hash: &[u8], pk: &PublicKey) -> bool {
+        self.0.verify_hashed(msg_hash, pk)
+    }
+
+    /// Returns the underlying signature.
+    pub fn as_raw(&self) -> &RawSignature {
+        &self.0
+    }
+}
+
+impl Default for Signature {
+    /// A "default" signature is a signature across an empty message by a secret key of 48 zeros.
+    fn default() -> Self {
+        let sk = match SecretKey::from_bytes(&[0; 48]) {
+            Ok(key) => key,
+            _ => unreachable!(), // Key is static, should not fail.
+        };
+        Signature(RawSignature::new(&[], &sk))
+    }
+}
+
+impl Encodable for Signature {
+    fn ssz_append(&self, s: &mut SszStream) {
+        s.append_vec(&self.0.as_bytes());
+    }
+}
+
+impl Decodable for Signature {
+    fn ssz_decode(bytes: &[u8], i: usize) -> Result<(Self, usize), DecodeError> {
+        let (sig_bytes, i) = decode_ssz_list(bytes, i)?;
+        let raw_sig = RawSignature::from_bytes(&sig_bytes).map_err(|_| DecodeError::TooShort)?;
```

Comment:
did we lose the `::TooLong` error case? (that was just in a comment)

Reply:
Yeah, I was thinking that a `TooLong` public key is not really a thing. If we get an error I can only really imagine it would be because its empty and therefore `TooShort`.

Ultimately, I'm just trying to pick an error from a basket of sub-optimal options... I might have a think about this and see if there's a better way to do this.